### ISSUE TITLE
[feature request] Please implement TencentARC/PhotoMaker

### ISSUE THREAD

**Issue by lanyusan** [lllyasviel/Fooocus#1980]
I recently played with the new release model and code named PhotoMaker by TencentARC.

The repo is here:

https://github.com/TencentARC/PhotoMaker

The result is very impressive. It does an extraordinary job of retaining features of reference face, better than ip adapter, while supporting stylization.

Can you please take a look?  @lllyasviel

Thanks.

**Comment by mashb1t** [comment]
Refers to https://github.com/lllyasviel/Fooocus/issues/1959

**Comment by fyepi** [comment]
@mashb1t 
Yes I get it that Fooocus isn't diffusers based. But the problem is that it seems the lots of people are building good stuff with diffusers. If Fooocus can't keep up with some of the most liked new features of diffusers ecosystem, people will move away.

**Comment by lllyasviel** [comment]
we will review some newer methods in the next round of dev.

Fooocus never "support" any features. if we add it, we will do some research and present something like Fooocus PhotoMaker to compete with all other avaliable software that only "support" those features. If we cannot make it better than all others, we will not add it. 

All features of Fooocus are unique and cannot be reproduced by other tools. 

Newer researches may outperform ip-adapter, but they may not outperform Fooocus Image Prompt.

thanks for your supppot for fooocus as always.

**Comment by BuBaLoM-JMP** [comment]
hi implement   https://github.com/csslc/ccsr?tab=readme-ov-file

**Comment by BuBaLoM-JMP** [comment]
Focus Best Generate Model THX

**Comment by lanyusan** [comment]
It is my pleasure to bring up some ideas to Fooocus team.  

My biggest hope is for Fooocus to catch up with Dalle 3 in prompt understanding. 

I have come across this repo recently. Hope the team can take a look as well.

LLaVA: Large Language and Vision Assistant

https://llava-vl.github.io/

Code:
https://github.com/haotian-liu/LLaVA

Demo:

https://github.com/LLaVA-VL/LLaVA-Interactive-Demo

> we will review some newer methods in the next round of dev.
> 
> Fooocus never "support" any features. if we add it, we will do some research and present something like Fooocus PhotoMaker to compete with all other avaliable software that only "support" those features. If we cannot make it better than all others, we will not add it.
> 
> All features of Fooocus are unique and cannot be reproduced by other tools.
> 
> Newer research may outperform ip-adapter, but they may not outperform Fooocus Image Prompt.
> 
> thanks for your supppot for fooocus as always.

**Comment by lanyusan** [comment]
Please also take a look at this. Its demo results look very good in following complex prompt. 

https://github.com/YangLing0818/RPG-DiffusionMaster

Abstract: RPG is a powerful training-free paradigm that can utilize proprietary MLLMs (e.g., GPT-4, Gemini-Pro) or open-source local MLLMs (e.g., miniGPT-4) as the prompt recaptioner and region planner with our complementary regional diffusion to achieve SOTA text-to-image generation and editing. Our framework is very flexible and can generalize to arbitrary MLLM architectures and diffusion backbones.


@mashb1t @lllyasviel

**Comment by machineminded** [comment]
I have made a simple integration in my fork of Fooocus using diffusers.  Could use some improvement but it is working well.  Not every scheduler has a mapping to a diffusers scheduler.
